pkg/codegen/go/gen.go

```diff
@@ -1702,4 +1705,19 @@ func getEnvOrDefault(def interface{}, parser envParser, vars ...string) interfac
 	}
 	return def
 }
+
+// PkgVersion uses reflection to determine the version of the current package.
+func PkgVersion() (semver.Version, error) {
+	type sentinal struct{}
+	pkgPath := reflect.TypeOf(sentinal{}).PkgPath()
+	re := regexp.MustCompile("^.*/pulumi-%s/sdk/v(\\d+)*")
+	if match := re.FindStringSubmatch(pkgPath); match != nil {
+		vStr := match[1]
+		if len(vStr) == 0 {
+			return semver.Version{Major: 1}, nil
+		}
+		return semver.MustParse(fmt.Sprintf("%%s.0.0", vStr)), nil
```

Comment:
This only gets the major version number right? Is that what we need?

Reply:
Yes, this is a simpler approach than the one you and I discussed yesterday. Eventually, we'd like to statically embed full package versions, but @pgavlin thought that approach was too risky of a change for the short term.

Instead, this method just uses the module version, which is only the major version component.

Reply:
👍🏾

Reply:
Will this work for v1 packages? I don't think that we emit a `v1` as part of the package path, do we?

Reply:
> Will this work for v1 packages? I don't think that we emit a `v1` as part of the package path, do we?

Yes, although I perhaps should add a comment. The conditional here handles that. https://github.com/pulumi/pulumi/pull/6117/files#diff-077694b97192c663e9314d22e695e6040d200174f3a2ef437021384302b9fd99R1716-R1718

Reply:
How will we hit that conditional? If the package path we emit doesn't include a `v1`, won't the regex fail to match at all?

Reply:
See e.g. https://regex101.com/r/nNxYDs/1

Reply:
Whoops, you're right. I moved the `v` out of the version capture group and forgot that it doesn't work for v1 since that wouldn't be present at all. Will update

Reply:
Fixed in https://github.com/pulumi/pulumi/pull/6121/commits/986ff7e81172d67dc03d043847a007bc31e7d5b9